packages/nextjs/components/scaffold-eth/Contract/ContractUI.tsx

```diff
@@ -84,6 +86,27 @@ export const ContractUI = ({ className = "", initialContractData }: ContractUIPr
     setAbi(selectedMethods);
   }, [initialContractData.abi]);
 
+  const {
+    data: contractNameData,
+    isLoading,
+    isError,
+  } = useContractRead({
+    address: initialContractData.address,
+    abi: initialContractData.abi,
+    chainId: mainChainId,
+    functionName: "name",
+  });
+
+  useEffect(() => {
+    if (isLoading) {
+      setContractName("Loading...");
+    } else if (isError) {
+      setContractName("Contract");
+    } else if (contractNameData && typeof contractNameData === "string") {
+      setContractName(contractNameData);
+    }
+  }, [contractNameData, isLoading, isError]);
```

Comment:
NITPICK : We can actually remove `useEffect` and also extra `contracName` state, since `data` returned by wagmi's `useContractRead` is reactive and will render component once it change, we could do this: 

```ts
const displayContractName = useMemo(() => {
  if (isLoading) return "Loading...";
  if (contractNameData && typeof contractNameData === "string") {
    return contractNameData;
  }
  // Default to "Contract" for errors or any other cases
  return "Contract";
}, [isLoading, contractNameData]);
```

Also maybe rename `isLoading` => `isContractNameLoading` 😅 just incase for future and we could remove `isError` since anywhich way we want default to be "Contract" 🙌

Reply:
Thanks a lot for the suggestion, love every lesson from your reviews ♥🔥 Just pushed the change 🙌